packages/turf-nearest-point-on-line/index.ts

```diff
@@ -206,62 +200,80 @@ function nearestPointOnSegment(
   const B = lngLatToVector(posB); // ... to posB
   const C = lngLatToVector(posC); // ... to posC
 
-  // Components of target point.
-  const [Cx, Cy, Cz] = C;
+  // The axis (normal vector) of the great circle plane containing the line segment
+  const segmentAxis = cross(A, B);
+
+  // Two degenerate cases exist for the segment axis cross product. The first is
+  // when vectors are aligned (within the bounds of floating point tolerance).
+  // The second is where vectors are antipodal (again within the bounds of
+  // tolerance. Both cases produce a [0, 0, 0] cross product which invalidates
+  // the rest of the algorithm, but each case must be handled separately:
+  // - The aligned case indicates coincidence of A and B. therefore this can be
+  //   an early return assuming the closest point is the end (for consistency).
+  // - The antipodal case is truly degenerate - an infinte number of great
+  //   circles are possible and one will always pass through C. We assume this
+  //   is undefined behavior and therefore throw. Callers can catch this and
+  //   return 0 if they wish.
+  if (segmentAxis[0] === 0 && segmentAxis[1] === 0 && segmentAxis[2] === 0) {
+    if (dot(A, B) > 0) {
+      return [[...posB], true];
+    } else {
+      throw new Error(
```

Comment:
I'm not sure I fully understand this part.
the bottom line from my perspective is that I don't expect nearest-point-on-line to ever throw.
So if this is caused by ambiguity (if I understand what you wrote above) then I would expect this to return the first one, even if it's not deterministic, I would still prefer that over a method that throws...
I might have completely misunderstood what you wrote, but I also couldn't see a test to over this code flow, so do let me know what I'm missing.

Reply:
If someone needs to know that this might be problematic you could export a method called isAntiPodal to allow testing it if someone is concerned about the results of this method. 
This avoid throwing and allows to decide what to do in this case if this interested you. 
Another option is simply to call it out in the method docs so that people might know this is a "know limitation".